jaeger-core/src/main/java/com/uber/jaeger/utils/Utils.java

```diff
@@ -50,7 +50,7 @@ public static int ipToInt(String ip) throws EmptyIpException, NotFourOctetsExcep
 
     int intIp = 0;
     for (byte octet : octets.getAddress()) {
-      intIp = (intIp << 8) | (octet);
+      intIp = (intIp << 8) | (octet & 0xFF);
```

Comment:
`octet` is promoted to integer via binary numeric promotion (https://docs.oracle.com/javase/specs/jls/se7/html/jls-5.html#jls-5.6.2), but it's a signed integer so the mask is filled with 1s on the left for negative numbers. `& 0xFF` converts it to unsigned (It's a byte so no need to care about int overflow. If it mattered then it should be 0xFFL). At least that's how I understand that.

In Java 8 that would instead be: `Byte.toUnsignedInt(octet)`